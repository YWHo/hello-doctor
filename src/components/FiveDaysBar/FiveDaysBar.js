import React from 'react';
import styled from 'styled-components';
import moment from 'moment';

const Container = styled.div`
  background-color: #3cb9c0;
  background-image: linear-gradient(to right, #3cb9c0, #2caaca);
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: start;
  height: 118.5px;
  padding: 0 0.4rem;
`;

const DateCircle = styled.div`
  background-color: ${props => (props.selected ? '#177d91' : 'transparent')};
  border-radius: 50%;
  color: #fff;
  display: flex;
  align-items: center;
  justify-content: center;
  margin-top: 8px;
  height: 66px;
  width: 66px;
`;

const DateCombined = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  height: 42px;
  width: 28px;
`;

const DateNum = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  height: 27px;
  font-size: 24px;
`;

const WeekDayName = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  height: 16px;
  font-size: 14px;
  font-weight: 300;
`;

export default function FiveDaysBar(props) {
  const { selectedDate = moment() } = props;

  const dateDiff = [-2, -1, 0, 1, 2];
  const dateList = dateDiff.map((num, idx) => {
    const tmpDate = moment(selectedDate).add(num, 'days');
    const selected = num === 0;
    return (
      <DateCircle key={`dateCi_${idx}`} selected={selected}>
        <DateCombined key={`dateCo_${idx}`}>
          <DateNum key={`dateNu_${idx}`}>{tmpDate.format('DD')}</DateNum>
          <WeekDayName key={`weekDa_${idx}`}>
            {tmpDate.format('ddd').toLowerCase()}
          </WeekDayName>
        </DateCombined>
      </DateCircle>
    );
  });

  return <Container>{dateList}</Container>;
}
