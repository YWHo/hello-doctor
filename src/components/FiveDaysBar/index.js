import FiveDaysBar from './FiveDaysBar';
export default FiveDaysBar;
