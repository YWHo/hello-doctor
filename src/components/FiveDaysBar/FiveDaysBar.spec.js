import React from 'react';
import moment from 'moment';
import FiveDaysBar from './FiveDaysBar';
import renderer from 'react-test-renderer';

describe('NavBar', () => {
  it('render correctly', () => {
    const component = renderer.create(
      <FiveDaysBar selectedDate={moment('20190619', 'YYYYMMDD')} />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
