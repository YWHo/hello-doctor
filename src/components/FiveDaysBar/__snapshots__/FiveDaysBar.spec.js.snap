// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar render correctly 1`] = `
<div
  className="sc-bdVaJa eLyiIX"
>
  <div
    className="sc-bwzfXH hBCumc"
    selected={false}
  >
    <div
      className="sc-htpNat hfnhPW"
    >
      <div
        className="sc-bxivhb eyjcHp"
      >
        17
      </div>
      <div
        className="sc-ifAKCX jyGSsx"
      >
        mon
      </div>
    </div>
  </div>
  <div
    className="sc-bwzfXH hBCumc"
    selected={false}
  >
    <div
      className="sc-htpNat hfnhPW"
    >
      <div
        className="sc-bxivhb eyjcHp"
      >
        18
      </div>
      <div
        className="sc-ifAKCX jyGSsx"
      >
        tue
      </div>
    </div>
  </div>
  <div
    className="sc-bwzfXH dLXFRF"
    selected={true}
  >
    <div
      className="sc-htpNat hfnhPW"
    >
      <div
        className="sc-bxivhb eyjcHp"
      >
        19
      </div>
      <div
        className="sc-ifAKCX jyGSsx"
      >
        wed
      </div>
    </div>
  </div>
  <div
    className="sc-bwzfXH hBCumc"
    selected={false}
  >
    <div
      className="sc-htpNat hfnhPW"
    >
      <div
        className="sc-bxivhb eyjcHp"
      >
        20
      </div>
      <div
        className="sc-ifAKCX jyGSsx"
      >
        thu
      </div>
    </div>
  </div>
  <div
    className="sc-bwzfXH hBCumc"
    selected={false}
  >
    <div
      className="sc-htpNat hfnhPW"
    >
      <div
        className="sc-bxivhb eyjcHp"
      >
        21
      </div>
      <div
        className="sc-ifAKCX jyGSsx"
      >
        fri
      </div>
    </div>
  </div>
</div>
`;
